ddb-popper/ddb-popper.js

```diff
@@ -0,0 +1,200 @@
+    let ddbPopper = new DDBPopper();
+});
+
+class DDBPopper {
+    constructor(){
+        this._hookRenderActorSheet();
+        this.existingPopup = null;
+    }
+
+    /**
+     * Some static config used elsewhere in the module
+     * @returns {Object} module config
+     */
+    static get CONFIG() {
+        return {
+            moduleName: "ddb-popper",
+            ddbLogo: "modules/ddb-popper/icons/dnd-beyond-b-red.png",
+            imgStyle: `display:inline-block;vertical-align:middle;height:16px`,
+            windowFeatures: "resizeable,scrollbars,location=no,width=768,height=968"
+        }  
+    }
+
+    /**
+     * Metadata used to register settings
+     * @returns {Object} Settings Metadata
+     */
+    static get SETTINGS_META() {
+        return {
+            name: "ddbCharacterURL",
+            type: String,
+            scope: "world",
+            default: ""
+        }
+    }
+
+    /**
+     * Hooks on render of ActorSheet5eCharacter in order to insert the DDB Button
+     */
+    _hookRenderActorSheet() {
+        Hooks.on("renderActorSheet5eCharacter", (app, html, data) => {
+            this._addDDBButton(html, data);
+        });
+    }
+
+    /**
+     * For a given actor's data, get the D&D Beyond URL if it exists in settings, 
+     * and if not, register and try again
+     * @param {Object} data -- an actor's data property
+     * @returns {String} ddbURL -- the matching D&D Beyond URL (if any)
+     */
+    async _getActorDDBURL(data) {
+        let ddbURL;
+
+        try {
+            ddbURL = await game.settings.get(DDBPopper.CONFIG.moduleName, data.actor._id);
+        } catch (e) {
+            if(e.message == "This is not a registered game setting") {
+                await game.settings.register(DDBPopper.CONFIG.moduleName, data.actor._id, DDBPopper.SETTINGS_META);
+                ddbURL = await game.settings.get(DDBPopper.CONFIG.moduleName, data.actor._id);
+            } else {
+                ddbURL = null;
+                throw(e);
+            }
+        } finally {
+            return ddbURL;
+        }
+    }
+
+    /**
+     * Adds the DDB button to the supplied html
+     * @param {Object} html -- the html of the ActorSheet5eCharacter
+     * @param {Object} data -- the data of the actor
+     */
+    async _addDDBButton (html, data) {
+        /**
+         * Finds the header and the close button
+         */
+        const windowHeader = html.parent().parent().find(".window-header");
+        const windowCloseBtn = windowHeader.find(".close");
+
+        /**
+         * jquery reference to the D&D Beyond button to add to the sheet
+         */
+        const ddbButton = $(
+            `<a class="ddb-popup">
+                <img src=${DDBPopper.CONFIG.ddbLogo} style=${DDBPopper.CONFIG.imgStyle}> DDB
+            </a>`
+        );
+        
+        /**
+         * Remove any existing instances of the button on the sheet
+         */
+        windowHeader.find('.ddb-popup').remove();
```

Comment:
If it exists already, why do you need to add it again?

Reply:
TBH I copied that piece from here: https://github.com/sillvva/foundry-vtt-modules/blob/3fbd15d0792229b22aa9f24359ba40c6af49ecf3/ddb-importer/ddb-importer.js#L27 the main reason would be ensuring that the button isn't duplicated? Or that it has the correct functionality tied in.

Reply:
In your case, you are controlling the adding of th button though? So it shouldn't be duplicated. This is definitely just an optimization change, buy I'm just curious if you saw any bug that caused you to do this.